src/theory/ext_theory.h

```diff
@@ -0,0 +1,236 @@
+/*********************                                                        */
+/*! \file ext_theory.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2017 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Base of the extended theory interface.
+ **
+ ** Base of the extended theory interface.
```

Comment:
Can you summarize in one or two sentences what this class is about? You can copy paste it to the cpp header.